### PR TITLE
Disallow GENERIC queue system

### LINKED ISSUE #8487: Ert 10.2 allows GENERIC as QUEUE_SYSTEM
The following configuration line is currently valid:

```
QUEUE_SYSTEM GENERIC
```
(add to poly.ert f.e.x)

Ert 10.1.9 will give an error on this
```
poly_gen.ert: Line 1 (Column 14-21): 'QUEUE_SYSTEM' argument 1 must be one of ['LSF', 'LOCAL', 'TORQUE', 'SLURM'] was 'GENERIC'
```

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #8487 

**Approach**
Split QueueSystem enum into 2 variants.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [x] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/unit_tests/config/test_queue_config.py [hunk 1]
```diff
@@ -67,18 +67,23 @@ def test_project_code_is_not_overwritten_if_set_in_config(queue_system):
...
+def test_that_an_invalid_queue_system_provided_raises_validation_error(
+    invalid_queue_system,
+):
+    """There is actually a "queue-system" called GENERIC, but it is
+    only there for making queue options global, it should not be
+    possible to try to use it"""
     filename = "config.ert"
 
     with open(filename, "w", encoding="utf-8") as f:
-        f.write(f"NUM_REALIZATIONS {num_real}\nQUEUE_SYSTEM VOID\n")
```
Comment (berland) [inline tests/unit_tests/config/test_queue_config.py:70]: num_real is taken out of the test, as it is pointless. Split out into a separate commit or ok?

Comment (berland) [reply tests/unit_tests/config/test_queue_config.py:75]: split.

**Comment by xjules** [comment]
Plenty of errors:
```
 src/ert/scheduler/__init__.py:23: error: "QueueOptions" has no attribute "driver_options"  [attr-defined]
```

**Comment by berland** [comment]
> Plenty of errors:
> 
> ```
>  src/ert/scheduler/__init__.py:23: error: "QueueOptions" has no attribute "driver_options"  [attr-defined]
> ```

Fixed.
